t/e2e/transactions/transactions_test.go

```diff
@@ -139,8 +141,52 @@ func (s *TransactionsTestSuite) TestCallRPCSendTransactionUpstream() {
 	s.Equal(`{"jsonrpc":"2.0","id":1,"result":"`+txHash.String()+`"}`, result)
 }
 
-// FIXME(tiabc): Sometimes it fails due to "no suitable peers found".
+// TestSendContractCompat tries to send transaction using the legacy "Data"
+// field, which is supported for backward compatibility reason.
+func (s *TransactionsTestSuite) TestSendContractTxCompat() {
+	initFunc := func(byteCode []byte, args *common.SendTxArgs) {
+		args.Data = (hexutil.Bytes)(byteCode)
```

Comment:
If I comment out this assignment, the e2e test still passes. Can you check it out?

Reply:
Yeah, our e2e tests don't really check what is passed in `Data` or `Input` and also it is perfectly valid for these fields to be empty...
Hmm, we really need more tests for RPC calls somewhere...

Reply:
It's valid for these fields to be empty actually. When a transaction is an ether exchange transaction, they are empty.